cpmdetail.py

```diff
@@ -25,41 +23,62 @@ def get_matches(puuid):
 
 
     def get_match_detail(matches, puuid):
-        matches_dict = {}
+        detailed_matches_dict = {}
         for match in matches:
             match_id = match
-            data = service.riotapi.get_match_detail(match_id, api_key)
-            data['puuid'] = puuid
-            participants = data.get("info").get("participants")
-            participant = [item for (index, item) in enumerate(participants) if item.get("puuid") == puuid]
-            participant_detail = {k: v for e in participant for (k, v) in e.items()}
-
-            match_detail = models.match_detail.MatchDetails(data)
-
-            match_participant_detail = models.match_detail.ParticipantMatchDetails(participant_detail)
-
-            match_id_dict = models.match_detail.merge(match_detail,match_participant_detail)
-
-            matches_dict[match_id] = match_id_dict
-
-        return matches_dict
-
-
-    def get_matches_ux(matches, puuid):
-        return service.riotapi.get_matches_ux(matches, puuid, api_key)
-
-
-    def get_participant(match_id, puuid):
-        # get participant id
-        return service.riotapi.get_users_participant_id(match_id, puuid, api_key)
+            match_detail_dict = service.riotapi.get_match_detail(match_id, api_key)
+            match_detail_dict['puuid'] = puuid
+            participants_dict = match_detail_dict.get("info").get("participants")
+            participant_dict = [participant for (index, participant) in enumerate(participants_dict) if
+                                participant.get("puuid") == puuid]
+            participant_detail_dict = {k: v for participant_dict in participant_dict for (k, v) in
+                                       participant_dict.items()}
+            match_detail_dict = models.match_detail.merge(models.match_detail.MatchDetails(match_detail_dict),
+                                                          models.match_detail.ParticipantMatchDetails(
+                                                              participant_detail_dict))
+            detailed_matches_dict[match_id] = match_detail_dict
+
+        return detailed_matches_dict
+
+
+    def get_index_post():
+        form = NameForm()
+        if form.validate_on_submit():
+            game_tag = form.gametag.data
```

Comment:
I would put forms and getting data from forms in the responsibility of index.py

Index.py wants to be able to ask cpmdetail.py "based on the info I received, given me this new info" and I think this change is making index.py little more than a wrapper around cpmdetail which idt is what we want.

Reply:
I wanted to, but didn't find a quick solution last night to accomplish this task. I tried to return the items as str and object. I didn't want to back it all out at that time though.